src/plugins/irc-events/link.js

```diff
@@ -406,6 +406,10 @@ function fetch(uri, headers) {
 						if (contentLength > limit) {
 							gotStream.destroy();
 						}
+						// If file is not to be stored we don't need to download further either
+						else if (!Helper.config.prefetchStorage) {
```

Comment:
I'd prefer to see these two branches joined since they have the same body. ie:

```js
if (contentLength > limit || !Helper.config.prefetchStorage) {
    gotStream.destroy();
}
```